pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/nullvalue/NullValueIndexType.java

```diff
@@ -77,11 +81,24 @@ public String getPrettyName() {
 
   @Override
   public ColumnConfigDeserializer<IndexConfig> createDeserializer() {
-    return IndexConfigDeserializer.ifIndexingConfig(
-        IndexConfigDeserializer.alwaysCall((TableConfig tableConfig, Schema schema) ->
-            tableConfig.getIndexingConfig().isNullHandlingEnabled()
-                ? IndexConfig.ENABLED
-                : IndexConfig.DISABLED));
+    return (TableConfig tableConfig, Schema schema) -> {
```

Comment:
> That is not correct. There is a special case for null value vector in [SegmentColumnarIndexCreator](https://github.com/apache/pinot/blob/55f29fe04dc7ed0ff18737fcec5b272969d6aba1/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java#L220) that creates the index depending on config.isNullHandlingEnabled without actually reading index configuration.

Does that mean we create null value vector for all columns only when it is configured in the table config, and the column level nullable is ignored; When reading the null value vector, we do check the column level nullable flag?
This behavior seems inconsistent and there is no way to only create null value vector for part of the columns. Can we make the write path logic the same as read path? i.e. check if it is enabled at column level, then decide whether to create the null value vector?

Reply:
Yesterday @walterddr and I were discussing about this PR. We think that https://github.com/apache/pinot/pull/11935 is a better approach given it doesn't need to modify how to load the index. The changes there are very trivial (changing from storing `boolean _enableNullHandling` with `NullMode _nullMode`). The main issue is that the null handling code is not well organized and very repetitive, so I ended up modifying more than 1k lines.

But we think we are going to be focused on that PR instead of this one. What are your thoughts?